source/materials/OpticalMaterialProperties.cc

```diff
@@ -736,300 +580,94 @@ G4MaterialPropertiesTable* OpticalMaterialProperties::LXe()
   LXe_mpt->AddConstProperty("YIELDRATIO", 0.03);
   LXe_mpt->AddConstProperty("ATTACHMENT", 1000.*ms);
 
-  G4double energy[2] = {0.01*eV, 100.*eV};
-  G4double abslen[2] = {1.e8*m, 1.e8*m};
-  LXe_mpt->AddProperty("ABSLENGTH", energy, abslen, 2);
-
-  return LXe_mpt;
-}
-
-G4MaterialPropertiesTable* OpticalMaterialProperties::FakeLXe()
-{
-  XenonLiquidProperties LXe_prop;
-  G4MaterialPropertiesTable* LXe_mpt = new G4MaterialPropertiesTable();
-
-  const G4int ri_entries = 18;
-  G4double ri_energy[ri_entries]
-    = {1*eV, 2*eV, 3*eV, 4*eV, 5*eV, 6*eV, 6.2*eV, 6.4*eV, 6.6*eV, 6.8*eV,
-       7*eV, 7.2*eV, 7.4*eV, 7.6*eV, 7.8*eV, 8*eV, 8.2*eV, 8.21*eV};
-
-  G4double rindex[ri_entries];
-
-  for (G4int i=0; i<ri_entries; i++) {
-    // rindex[i] = LXe_prop.RefractiveIndex(ri_energy[i]);
-    rindex[i] = 1.7;
-    //   G4cout << ri_energy[i] << ", " << rindex[i] << G4endl;
-  }
-
-
-
-  LXe_mpt->AddProperty("RINDEX", ri_energy, rindex, ri_entries);
-  //LXe_mpt->AddProperty("FASTCOMPONENT", sc_energy, intensity, sc_entries);
-  // LXe_mpt->AddProperty("ELSPECTRUM", sc_energy, intensity, sc_entries);
-  //  LXe_mpt->AddProperty("SLOWCOMPONENT", sc_energy, intensity, sc_entries);
-  // LXe_mpt->AddConstProperty("SCINTILLATIONYIELD", 58708./MeV);
-  LXe_mpt->AddConstProperty("RESOLUTIONSCALE", 1);
-  LXe_mpt->AddConstProperty("RAYLEIGH", 36.*cm);
-  // check constants with the Aprile
-  //  LXe_mpt->AddConstProperty("FASTTIMECONSTANT",2.2*ns);
-  // LXe_mpt->AddConstProperty("SLOWTIMECONSTANT",40.*ns);
-  //  LXe_mpt->AddConstProperty("ELTIMECONSTANT", 50.*ns);
-  //LXe_mpt->AddConstProperty("YIELDRATIO", .1);
-  LXe_mpt->AddConstProperty("ATTACHMENT", 1000.*ms);
+  std::vector<G4double> abs_energy = {optPhotMinE_, optPhotMaxE_};
+  std::vector<G4double> abs_length = {noAbsLength_, noAbsLength_};
 
-  G4double energy[2] = {0.01*eV, 100.*eV};
-  G4double abslen[2] = {1.e8*m, 1.e8*m};
-  LXe_mpt->AddProperty("ABSLENGTH", energy, abslen, 2);
+  assert(abs_energy.size() == abs_length.size());
+  LXe_mpt->AddProperty("ABSLENGTH", abs_energy.data(), abs_length.data(), abs_energy.size());
 
   return LXe_mpt;
 }
 
+
 G4MaterialPropertiesTable* OpticalMaterialProperties::LXe_nconst()
 {
   XenonLiquidProperties LXe_prop;
   G4MaterialPropertiesTable* LXe_mpt = new G4MaterialPropertiesTable();
 
-  const G4int ri_entries = 18;
-  G4double ri_energy[ri_entries]
-    = {1*eV, 2*eV, 3*eV, 4*eV, 5*eV, 6*eV, 6.2*eV, 6.4*eV, 6.6*eV, 6.8*eV,
-       7*eV, 7.2*eV, 7.4*eV, 7.6*eV, 7.8*eV, 8*eV, 8.2*eV, 8.21*eV};
+  std::vector<G4double> ri_energy = {optPhotMinE_, optPhotMaxE_};
+  std::vector<G4double> ri_index  = {1.7, 1.7};
```

Comment:
The refractive index at the scintillation peak is 1.69. This property was used years ago for some studies and `1.7` was used (a worse approximation, but still similar to the most probable value). I hadn't changed it, because this property is not used anymore, but I will, since I agree that `1.69` is more accurate.